split-webapp/split/src/pages/Split/index.jsx

```diff
@@ -58,18 +74,31 @@ class Split extends Component {
       byId: {}
     };
 
-    for (let i = 0; i < MINIMUM_PEOPLE_COUNT; i += 1) {
+    // Populate fields with values of existing bill
+    let users = [];
+    let cost = 0;
+    let title = "";
+    if (this.editBill) {
+      users = this.editBill.payments.map(person => person.from);
+      users.push(this.editBill.payments[0].to); 
+      cost = this.editBill.total;
+      title = this.editBill.title;
+    }
+
+    const userLength = users.length > 0 ? users.length : MINIMUM_PEOPLE_COUNT;
+    for (let i = 0; i < userLength; i += 1) {
       const userId = uuidv4();
       initialUsers.allIds.push(userId);
-      initialUsers.byId[userId] = { name: "" };
+      initialUsers.byId[userId] = { name: users.pop() || "" };
```

Comment:
Is it fine that `shift()` is a slower than `pop()` since `shift()` requires each element in the array to re-index? The difference should really only be noticeable for very large bills though.

Reply:
Yeah, I wouldn't worry too much about the performance of `shift()`

- I don't expect the user to input more than 100 people (I hope 😦)
- The processing done here is one-off when the split bill page is loaded, so the user wouldn't notice much
- JS engines are pretty smart at optimizing

...unless if you do notice some performance problems though.